static/elements/chromedash-featurelist.html

```diff
@@ -130,38 +178,76 @@
+          // Array of matches, each an array of the form:
+          // [ full match, quoted-string-or-word, contents-of-quoted-string,
+          //   word, separator ].
+          var reMatches = [];
+          var match;
+          while ((match = regExp.exec(val)) !== null) {
+            reMatches.push(match);
+          }
+
+          // Query parts of the form: "name : value".
+          var propertyQueries = [];
+          // Query parts of the form: "name = value", "name < value", etc.
+          var operatorQueries = [];
+          // Other words in the query (not matching forms above).
+          var keywordQueries = [];
+
+          // Accumulate keyword and property queries.
+          var i;
+          for (i = 0; i < reMatches.length; i++) {
+            match = reMatches[i];
+            var part = match[2] || match[3];
+            var sep = match[4];
+            var nextPart = i < reMatches.length - 1 ?
+                           (reMatches[i + 1][2] || reMatches[i + 1][3]) : null;
+            if (sep && sep.trim().match(/^:$/) !== null && nextPart !== null) {
+              // Separator is ":", and there exists a right-hand-side.
+              // Store property query: "propertyName : propertyValue".
+              propertyQueries.push([part, nextPart]);
+              i++;
+            } else if (sep && sep.trim().match(/^(<=|>=|==|<|>|=)$/) !== null &&
+                       nextPart !== null) {
+              // Separator is an ordered comparison operator and there exists a
+              // right-hand-side.
+              // Store operator query "name <<operator>> value".
+              operatorQueries.push([part, sep, nextPart]);
+              i++;
+            } else {
+              // No special operator found. Store non-separator part as keyword
+              // query.
+              keywordQueries.push(part);
+            }
+          }
+
+          // Construct a list filter for each query part, and store them all in
+          // a list.
+          var filters = propertyQueries.map(
+            this._getPropertyFilter.bind(this)).concat(
+              operatorQueries.map(
+                this._getOperatorFilter.bind(this))).concat(
+                  keywordQueries.map(
+                    this._getKeywordFilter.bind(this)));
+
+          // Apply this.filtered = this.features filtered-by filters.
+          if (filters.length === 0) {
+            this.filtered = this.features;
           } else {
-            // Only one of the platforms needs to match.
-            // Do a general search over misc fields.
-            // Remove "=" prefix so "in development"/"proposed" version queries
-            // return results.
-            if (val.indexOf('=') == 0) {
-              val = val.substring(1);
+            var results = this.features;
+            for (i = 0; i < filters.length; i++) {
```

Comment:
Switched to `function(...) { ... for (var i...) { ... } ... for (var i...) { ... } ... }` (harmless variable redeclaration) as per discussion.